internal/controller/pipelinerun_controller_test.go

```diff
@@ -14,19 +14,138 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// What to test:
+// - check countRunningPipelineRuns
+// - with a queue of 3 pipelineruns, when one finishes one other starts and one stays pending
+
 package controller
 
 import (
+	"strconv"
+
 	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	local_tekton "github.com/konflux-ci/mintmaker/internal/pkg/tekton"
+	. "github.com/konflux-ci/mintmaker/pkg/common"
 )
 
 var _ = Describe("PipelineRun Controller", func() {
-	Context("When reconciling a resource", func() {
+	Context("When reconciling pipelineruns", func() {
+
+		originalMaxSimultaneousPipelineRuns := MaxSimultaneousPipelineRuns
+
+		_ = BeforeEach(func() {
+
+			MaxSimultaneousPipelineRuns = 2
+
+			for i := range 3 {
+				pplrName := "pplnr" + strconv.Itoa(i)
+				pipelineRunBuilder := local_tekton.NewPipelineRunBuilder(pplrName, MintMakerNamespaceName)
+				pipelinerun, err := pipelineRunBuilder.Build()
+				Expect(err).NotTo(HaveOccurred())
+				k8sClient.Create(ctx, pipelinerun)
```

Comment:
You should check this call result:
```
Expect(k8sClient.Create(ctx, pipelinerun)).Should(Succeed())
```
with this, you will see the pipelineruns are not created successfully, due to "mintmaker" namespace not exists, so you should create the namespace first:
```
createNamespace(MintMakerNamespaceName)
```

Reply:
Good catch! This fixed the tests indeed, thanks a lot!! 🥳